partcad/src/partcad/ai_feature_file.py

```
@@ -0,0 +1,92 @@
+        self.filename = None
+
+        # Switch to lowercase to normalize the file extension in the URL
+        lower_url = url.lower()
+        if lower_url.startswith(("http://", "https://")):
+            response = requests.get(self.url, allow_redirects=True, timeout=15)
+            if response.status_code == 200:
+                self.filename = tempfile.NamedTemporaryFile(suffix=self.suffix, delete=True, delete_on_close=False)
+                self.filename.write(response.content)
+                self.filename.close()
+            else:
+                raise Exception(f"Failed to download the file {self.url}")
```

Ruff (0.8.2)

Create your own exception

(TRY002)